src/core/Global.ttslua

```diff
@@ -456,358 +268,281 @@ function drawChaostoken(params)
     end
-end
 
-function spawnToken_2(id, pos)
-    local url = getImageUrl(id)
-    if url ~= '' then
-        local obj = spawnObject({
-            type = 'Custom_Tile',
-            position = {pos.x, pos.y + 3, pos.z},
-            rotation = {x = 0, y = 260, z = 0}
-                    })
-        obj.setCustomObject({
-            type = 2,
-            image = url,
-            thickness = 0.10,
-                    })
-        obj.scale {0.81, 1, 0.81}
-        obj.setName(getTokenName({ url=url }))
-        return obj
+    if data.hard ~= nil then
+      buttonParameters.label = "Hard"
+      buttonParameters.click_function = "hardClick"
+      args.object.createButton(buttonParameters)
+      buttonParameters.position[3] = buttonParameters.position[3] + 0.20
     end
-end
 
-function getTokenName(params)
-    local name = IMAGE_TOKEN_MAP[params.url]
-    if name == nil then name = "" end
-    return name
-end
-
-function getImageUrl(id)
-    if id == 'p1' then return 'https://i.imgur.com/uIx8jbY.png' end
-    if id == '0' then return 'https://i.imgur.com/btEtVfd.png' end
-    if id == 'm1' then return 'https://i.imgur.com/w3XbrCC.png' end
-    if id == 'm2' then return 'https://i.imgur.com/bfTg2hb.png' end
-    if id == 'm3' then return 'https://i.imgur.com/yfs8gHq.png' end
-    if id == 'm4' then return 'https://i.imgur.com/qrgGQRD.png' end
-    if id == 'm5' then return 'https://i.imgur.com/3Ym1IeG.png' end
-    if id == 'm6' then return 'https://i.imgur.com/c9qdSzS.png' end
-    if id == 'm7' then return 'https://i.imgur.com/4WRD42n.png' end
-    if id == 'm8' then return 'https://i.imgur.com/9t3rPTQ.png' end
-    if id == 'skull' then return 'https://i.imgur.com/stbBxtx.png' end
-    if id == 'cultist' then return 'https://i.imgur.com/VzhJJaH.png' end
-    if id == 'tablet' then return 'https://i.imgur.com/1plY463.png' end
-    if id == 'elder' then return 'https://i.imgur.com/ttnspKt.png' end
-    if id == 'red' then return 'https://i.imgur.com/lns4fhz.png' end
-    if id == 'blue' then return 'https://i.imgur.com/nEmqjmj.png' end
-	if id == 'frost' then return 'http://cloud-3.steamusercontent.com/ugc/1858293462583104677/195F93C063A8881B805CE2FD4767A9718B27B6AE/' end
-    return ''
-end
+    if data.expert ~= nil then
+      buttonParameters.label = "Expert"
+      buttonParameters.click_function = "expertClick"
+      args.object.createButton(buttonParameters)
+      buttonParameters.position[3] = buttonParameters.position[3] + 0.20
+    end
 
-function cleanContainer(container)
-    for _, item in ipairs(container.getObjects()) do
-        destroyObject(container.takeObject({}))
+    if data.standalone ~= nil then
+      buttonParameters.label = "Standalone"
+      buttonParameters.click_function = "standaloneClick"
+      args.object.createButton(buttonParameters)
     end
+  end
 end
 
-function getObjectsInZone(zoneId)
-    local zoneObject = getObjectCache(zoneId)
+function fillContainer(args)
```

Comment:
initializeChaosBag?

Especially in Global (which can and does have a wide variety of things it does) generic names have no context to work with.

At some point we should probably split these to independent files based on function, and require them...  but that's another day.

Reply:
same as above - not easily changeable because it is referenced by every difficulty selector

Reply:
That makes sense.  Can we at least enhance the method doc here then?  Describe (at least at a high level) what it does, where it's used, and what the parameters expected in the args table are?

Reply:
Documentation added